.travis.yml

```diff
@@ -5,9 +5,23 @@ python:
   - "3.3"
   - "3.4"
   - "3.5"
+  - "pypy"
 
 sudo: false
 install:
+  - |
+        if [ "$TRAVIS_PYTHON_VERSION" = "pypy" ]; then
+          export PYENV_ROOT="$HOME/.pyenv"
+          if [ -f "$PYENV_ROOT/bin/pyenv" ]; then
+            pushd "$PYENV_ROOT" && git pull && popd
+          else
+            rm -rf "$PYENV_ROOT" && git clone --depth 1 https://github.com/yyuu/pyenv.git "$PYENV_ROOT"
+          fi
+          export PYPY_VERSION="5.6.0"
+          "$PYENV_ROOT/bin/pyenv" install --skip-existing "pypy-$PYPY_VERSION"
+          virtualenv --python="$PYENV_ROOT/versions/pypy-$PYPY_VERSION/bin/python" "$HOME/virtualenvs/pypy-$PYPY_VERSION"
+          source "$HOME/virtualenvs/pypy-$PYPY_VERSION/bin/activate"
```

Comment:
So, this virtualenv thing isn't working.

The TL;DR is that Travis runs each new `-`-prefixed line in a new shell. This means that when you `source` the virtualenv here, the very next command loses all those environment variables.

To get this to work the way you want you'll need to use some shell scripts. [certitude uses a pair of install/run shell scripts](https://github.com/python-hyper/certitude/tree/master/.travis) that DTRT: you may want to copy them.